testinfra/common/test_fpf_apt_repo.py

```diff
@@ -15,9 +18,17 @@ def test_fpf_apt_repo_present(File):
     installed, e.g. for OSSEC. Install state for those packages
     is tested separately.
     """
-    f = File('/etc/apt/sources.list.d/apt_freedom_press.list')
-    assert f.contains('^deb \[arch=amd64\] https:\/\/apt\.freedom\.press '
-                      'trusty main$')
+
+    # If the var fpf_apt_repo_url test var is apt-test, validate that the
+    # apt repository is configured on the host
+    if test_vars.fpf_apt_repo_url == "https://apt-test.freedom.press":
+        f = File('/etc/apt/sources.list.d/apt_test_freedom_press.list')
+        assert f.contains('^deb \[arch=amd64\] '
+                          'https:\/\/apt-test\.freedom\.press trusty main$')
+    else:
+        f = File('/etc/apt/sources.list.d/apt_freedom_press.list')
+        assert f.contains('^deb \[arch=amd64\] https:\/\/apt\.freedom\.press '
+                          'trusty main$')
```

Comment:
Clunky logic, but you're right to implement it! We could start forcing a specific filepath for the apt repo, which would help with problems like #1145. Here, the only conditional is the filepath—we consolidate the assert statement to reference `re.escape(test_vars.fpf_apt_repo_url`.